stock_indicators/indicators/vwap.py

```
@@ -39,10 +40,26 @@
     """
-    if isinstance(start, int):
-        start = datetime(start, month, day, hour, minute)
-
-    results = CsIndicator.GetVwap[Quote](CsList(Quote, quotes), CsDateTime(start) if start else None)
+    # Backward compatibility: support positional year,month,day,hour as in tests
+    if legacy_date_parts:
+        # Interpret: start is actually the year; legacy_date_parts supply remaining
+        year = start if isinstance(start, int) else None
+        parts = list(legacy_date_parts)
+        # Fill provided parts into month, day, hour, minute if given positionally
+        if len(parts) > 0: month = parts[0]
```

Ruff (0.14.1)

Multiple statements on one line (colon)

(E701)